index.js

```diff
@@ -557,13 +542,13 @@ class HtmlWebpackPlugin {
 
     /**
      * @type {string} the configured public path to the asset root
-     * if the absolute path publicPath is set in the current webpack config use it otherwise
+     * if a path publicPath is set in the current webpack config use it otherwise
      * fallback to a realtive path
      */
-    let publicPath = this.isAbsolutePath(compilation.options.output.publicPath)
-      // If the absolute path is set in the publicPath use it
+    let publicPath = compilation.mainTemplate.getPublicPath({hash: compilationHash}).trim() !== ''
```

Comment:
Will `compilation.mainTemplate.getPublicPath({hash: compilationHash})` always return a string?

Maybe we could change it to:

```
const webpackPublicPath = compilation.mainTemplate.getPublicPath({hash: compilationHash});
const isPublicPathDefined = webpackPublicPath !== undefined && String(webpackPublicPath).trim() !== '';
```

That way we don't have to run `compilation.mainTemplate.getPublicPath({hash: compilationHash})` twice.

Reply:
`compilation.mainTemplate.getPublicPath({hash: compilationHash})` now always return a string. 

https://github.com/webpack/webpack/blob/8e841feda3ef32c64df1a175d0d9f71e47cfbe8e/lib/MainTemplate.js#L500

It will be right :)
```javascript
const isPublicPathDefined = webpackPublicPath.trim() !== '';
```